### ISSUE TITLE
Should "delete" key on vertex cut the vertex?

### ISSUE THREAD

**Issue by samreid** [phetsims/circuit-construction-kit-common#307]
While @jessegreenberg and I were working on https://github.com/phetsims/circuit-construction-kit-common/issues/114 we raised the question of whether the delete key should cut a vertex.

**Comment by arouinfar** [comment]
@samreid for a non-accessible version, I think using delete to cut a vertex would make sense, as it mirrors the interaction to delete a component. Given how close the cut button is to the vertex, I don't think this feature would be necessary for 1.0. 

For a11y, I don't know that we would use the delete key here. We've been mapping sim elements to native HTML elements, so buttons (such as the cut button) are activated with enter or space.

**Comment by emily-phet** [comment]
@samreid @arouinfar "Delete" key for a sim-specific use seems fine to me. We can list custom shortcut keys in the keyboard nav help popup. I don't know of any other expected behavior for the "Delete" key within an HTML page - so I don't think this would be overriding some other expected behavior.

**Comment by arouinfar** [comment]
SR estimates 2 hours to implement this feature. AP says go for it, but leave as an easter egg.

**Comment by jessegreenberg** [comment]
@samreid please let me know if I can assist with this.

**Comment by samreid** [comment]
I implemented this using the same pattern as we used for deleting circuit elements and it seems to work OK.  I tested for 1-2 minutes using Chrome only.

**Comment by samreid** [comment]
I published a version here for testing:
http://www.colorado.edu/physics/phet/dev/html/circuit-construction-kit-dc/1.0.0-dev.71/circuit-construction-kit-dc_en.html

While testing in Firefox, I noticed that the delete key (for circuit elements and vertices) causes the browser to navigate back a page.  @jessegreenberg do you know how to work around that?  I noticed that in the published version of John Travoltage that firefox doesn't navigate back when delete is pressed, so perhaps there is a flag for this, or I need to preventDefault somewhere?

**Comment by jessegreenberg** [comment]
Thanks @samreid, I tried published john-travoltage in Firefox and the backspace key does navigate to the previous page.  I will look into seeing if we can use preventDefault or some other trick to prevent this behavior.

**Comment by jessegreenberg** [comment]
```preventDefault``` seems to do the trick for Firefox on Windows, Ill give it a try on a few other platforms.

**Comment by jessegreenberg** [comment]
Seems to be working on MacOS 10.12.6 (beta) on Chrome, Firefox, and Safari. Ill commit the change.

**Comment by jessegreenberg** [comment]
Pushed.  @samreid I also noticed that a vertex can be removed when the cut button is disabled, perhaps 
```js
circuit.cutVertex( circuit.getSelectedVertex() );
```
should be hidden behind ```cutButton.enabled```?  I won't commit that in case you have a preferred method.

**Comment by samreid** [comment]
In the above commit, I'm only allowing the keyboard to cut if the cut button is enabled.  It seems like it is working great.  @jessegreenberg anything else for this issue?

**Comment by samreid** [comment]
@jessegreenberg it would be good to finish up this issue when you have time.

**Comment by jessegreenberg** [comment]
Sorry for the delay @samreid, there is nothing else that needs to be done that I am aware of. Thanks! Closing the issue.

**Comment by samreid** [comment]
Excellent, thanks!
